dask/bytes/core.py

```diff
@@ -0,0 +1,156 @@
+import io
+import sys
+
+from toolz import merge
+
+from .compression import seekable_files, files as cfiles
+from .utils import SeekableFile
+from ..delayed import delayed
+from ..utils import system_encoding
+
+delayed = delayed(pure=True)
+
+_read_bytes = dict()
+_open_files = dict()
+_open_text_files = dict()
+
+
+def read_bytes(path, delimiter=None, not_zero=False, blocksize=2**27,
+                     sample=True, compression=None, **kwargs):
+    """ Convert path to a list of delayed values
+
+    The path may be a filename like ``'2015-01-01.csv'`` or a globstring
+    like ``'2015-*-*.csv'``.
+
+    The path may be preceeded by a protocol, like ``s3://`` or ``hdfs://`` and,
+    if those libraries are installed, the futures will point to those locations
+    instead.
+
+    This cleanly breaks data by a delimiter if given, so that block boundaries
+    start directly after a delimiter and end on the delimiter.
+
+    Parameters
+    ----------
+    path: string
+    delimiter: bytes
+        An optional delimiter, like ``b'\n'`` on which to split blocks of bytes
+    not_zero: force seek of start-of-file delimiter, discarding header
+    blocksize: int (=128MB)
+        Chunk size
+    compression: string or None
+        String like 'gzip' or 'xz'.  Must support efficient random access.
+    sample: bool, int
+        Whether or not to return a sample from the first 10k bytes
+    **kwargs: dict
+        Options to send down to backend.  Includes authentication information
+        for systems like S3 or HDFS
+
+    Examples
+    --------
+    >>> sample, blocks = read_bytes('2015-*-*.csv', delimiter=b'\\n')  # doctest: +SKIP
+    >>> sample, blocks = read_bytes('s3://2015-*-*.csv', delimiter=b'\\n')  # doctest: +SKIP
+
+    Returns
+    -------
+    10kB sample header and list of ``dask.Delayed`` objects or list of lists of
+    delayed objects if ``fn`` is a globstring.
+    """
+    if '://' in path:
```

Comment:
Local files are typically written "file://..." to be explicit in the presence of remote protocols. Maybe "file" can be a protocol that is identical to "local".